src/test/java/org/opensearch/neuralsearch/query/HybridQueryPostFilterIT.java

```diff
@@ -0,0 +1,418 @@
+                List.of(),
+                List.of(),
+                List.of(INTEGER_FIELD_1),
+                List.of(INTEGER_FIELD_STOCK_4_VALUE),
+                List.of(KEYWORD_FIELD_2),
+                List.of(KEYWORD_FIELD_CATEGORY_2_VALUE),
+                List.of(),
+                List.of()
+            );
+
+            addKnnDoc(
+                indexName,
+                "5",
+                List.of(),
+                List.of(),
+                Collections.singletonList(TEXT_FIELD_1),
+                Collections.singletonList(TEXT_FIELD_NAME_5_VALUE),
+                List.of(),
+                List.of(),
+                List.of(INTEGER_FIELD_1),
+                List.of(INTEGER_FIELD_STOCK_5_VALUE),
+                List.of(KEYWORD_FIELD_2),
+                List.of(KEYWORD_FIELD_CATEGORY_1_VALUE),
+                List.of(),
+                List.of()
+            );
+
+            addKnnDoc(
+                indexName,
+                "6",
+                List.of(),
+                List.of(),
+                Collections.singletonList(TEXT_FIELD_1),
+                Collections.singletonList(TEXT_FIELD_NAME_6_VALUE),
+                List.of(),
+                List.of(),
+                List.of(INTEGER_FIELD_1),
+                List.of(INTEGER_FIELD_STOCK_5_VALUE),
+                List.of(KEYWORD_FIELD_2),
+                List.of(KEYWORD_FIELD_CATEGORY_3_VALUE),
+                List.of(),
+                List.of()
+            );
+        }
+    }
+
+    private HybridQueryBuilder createHybridQueryBuilderWithMatchTermAndRangeQuery(String text, String value, int lte, int gte) {
+        MatchQueryBuilder matchQueryBuilder = QueryBuilders.matchQuery(TEXT_FIELD_1, text);
+        TermQueryBuilder termQueryBuilder = QueryBuilders.termQuery(TEXT_FIELD_1, value);
+        RangeQueryBuilder rangeQueryBuilder = QueryBuilders.rangeQuery(INTEGER_FIELD_1).gte(gte).lte(lte);
+        HybridQueryBuilder hybridQueryBuilder = new HybridQueryBuilder();
+        hybridQueryBuilder.add(matchQueryBuilder).add(termQueryBuilder).add(rangeQueryBuilder);
+        return hybridQueryBuilder;
+    }
+
+    private HybridQueryBuilder createHybridQueryBuilderScenarioWithMatchAndRangeQuery(String text, int lte, int gte) {
+        MatchQueryBuilder matchQueryBuilder = QueryBuilders.matchQuery(TEXT_FIELD_1, text);
+        RangeQueryBuilder rangeQueryBuilder = QueryBuilders.rangeQuery(INTEGER_FIELD_1).gte(gte).lte(lte);
+        HybridQueryBuilder hybridQueryBuilder = new HybridQueryBuilder();
+        hybridQueryBuilder.add(matchQueryBuilder).add(rangeQueryBuilder);
+        return hybridQueryBuilder;
+    }
+
+    private QueryBuilder createPostFilterQueryBuilderWithRangeQuery(int lte, int gte) {
+        return QueryBuilders.rangeQuery(INTEGER_FIELD_1).gte(gte).lte(lte);
+    }
+
+    private QueryBuilder createPostFilterQueryBuilderWithBoolShouldQuery(String query, int lte, int gte) {
```

Comment:
```suggestion
    private QueryBuilder createQueryBuilderWithBoolShouldQuery(String query, int lte, int gte) {
```
It's a plain QueryBuilder, not specific to post filters

Reply:
same for other private methods below